diff -ur nodejs-lloyd_v1.orig/wscript nodejs-lloyd_v1/wscript
--- nodejs-lloyd_v1.orig/wscript	2010-05-10 16:07:32.000000000 -0600
+++ nodejs-lloyd_v1/wscript	2010-05-10 16:07:38.000000000 -0600
@@ -134,23 +134,10 @@
     if sys.platform.startswith("freebsd"):
       conf.fatal("Install the libexecinfo port from /usr/ports/devel/libexecinfo.")
 
-  if conf.check_cfg(package='openssl',
-                    args='--cflags --libs',
-                    uselib_store='OPENSSL'):
-    conf.env["USE_OPENSSL"] = True
-    conf.env.append_value("CXXFLAGS", "-DHAVE_OPENSSL=1")
-  else:
-    libssl = conf.check_cc(lib='ssl',
-                           header_name='openssl/ssl.h',
-                           function_name='SSL_library_init',
-                           libpath=['/usr/lib', '/usr/local/lib', '/opt/local/lib', '/usr/sfw/lib'],
-                           uselib_store='OPENSSL')
-    libcrypto = conf.check_cc(lib='crypto',
-                              header_name='openssl/crypto.h',
-                              uselib_store='OPENSSL')
-    if libcrypto and libssl:
-      conf.env["USE_OPENSSL"] = True
-      conf.env.append_value("CXXFLAGS", "-DHAVE_OPENSSL=1")
+  conf.env['HAVE_OPENSSL'] = True
+  conf.env['USE_OPENSSL'] = True
+  conf.env['LIB_OPENSSL'] = ['ssl_s', 'crypto_s', 'zlib_s']
+  conf.env['LIBPATH_OPENSSL'] = os.environ['BAKERY_LIBPATH']
 
   conf.check(lib='rt', uselib_store='RT')
 
