--- libarchive-2.7.902a/build/cmake/FindZLIB.cmake.orig	1969-12-31 16:00:00.000000000 -0800
+++ libarchive-2.7.902a/build/cmake/FindZLIB.cmake	2010-01-04 12:45:30.489257800 -0800
@@ -0,0 +1,30 @@
+# - Find zlib
+# Find the native zlib includes and library
+#
+#  ZLIB_INCLUDE_DIR    - where to find zlib.h, etc.
+#  ZLIB_LIBRARIES      - List of libraries when using libzlib_s
+#  ZLIB_FOUND          - True if libzlib_s found.
+
+IF (ZLIB_INCLUDE_DIR)
+  # Already in cache, be silent
+  SET(ZLIB_FIND_QUIETLY TRUE)
+ENDIF (ZLIB_INCLUDE_DIR)
+
+FIND_PATH(ZLIB_INCLUDE_DIR zlib.h
+	  PATHS ENV ZLIB_INC_DIR)
+FIND_LIBRARY(ZLIB_LIBRARY NAMES zlib_s 
+	  PATHS ENV EXT_LIB_DIR)
+
+# handle the QUIETLY and REQUIRED arguments and set ZLIB_FOUND to TRUE if 
+# all listed variables are TRUE
+INCLUDE(FindPackageHandleStandardArgs)
+FIND_PACKAGE_HANDLE_STANDARD_ARGS(ZLIB DEFAULT_MSG ZLIB_LIBRARY ZLIB_INCLUDE_DIR)
+
+IF(ZLIB_FOUND)
+  SET( ZLIB_LIBRARIES ${ZLIB_LIBRARY} )
+ELSE(ZLIB_FOUND)
+  SET( ZLIB_LIBRARIES )
+ENDIF(ZLIB_FOUND)
+
+
+MARK_AS_ADVANCED( ZLIB_LIBRARY ZLIB_INCLUDE_DIR )
