diff -ur libarchive-2.8.0.orig/CMakeLists.txt libarchive-2.8.0/CMakeLists.txt
--- libarchive-2.8.0.orig/CMakeLists.txt	2010-04-22 09:58:30.000000000 -0600
+++ libarchive-2.8.0/CMakeLists.txt	2010-04-22 10:03:44.000000000 -0600
@@ -46,8 +46,11 @@
 OPTION(ENABLE_CPIO_SHARED "Enable dynamic build of cpio" OFF)
 OPTION(ENABLE_XATTR "Enable extended attribute support" ON)
 OPTION(ENABLE_ACL "Enable ACL support" ON)
+OPTION(ENABLE_XML "Enable XML support if available" ON)
+OPTION(ENABLE_LZMA "Enable LZMA support if available" ON)
 OPTION(ENABLE_TEST "Enable unit and regression tests" ON)
 
+
 IF(ENABLE_TEST)
 	ENABLE_TESTING()
 ENDIF(ENABLE_TEST)
@@ -172,7 +175,12 @@
 #
 # Find LZMA
 #
-FIND_PACKAGE(LZMA)
+IF(ENABLE_LZMA)
+  FIND_PACKAGE(LZMA)
+ELSE(ENABLE_LZMA)
+  SET(LZMA_FOUND 0)
+  SET(LZMADEC_FOUND 0)
+ENDIF(ENABLE_LZMA)
 IF(LZMA_FOUND)
   SET(HAVE_LIBLZMA 1)
   SET(HAVE_LZMA_H 1)
@@ -336,30 +344,36 @@
 #
 # Find Libxml2
 #
-FIND_PACKAGE(LibXml2)
-IF(LIBXML2_FOUND)
-  INCLUDE_DIRECTORIES(${LIBXML2_INCLUDE_DIR})
-  LIST(APPEND ADDITIONAL_LIBS ${LIBXML2_LIBRARIES})
-  SET(HAVE_LIBXML2 1)
-  # libxml2's include files use iconv.h
-  # We need a directory path of iconv.h so that it won't fail to check
-  # "libxml/xmlreader.h".
-  FIND_PATH(ICONV_INCLUDE_DIR iconv.h)
-  SET(CMAKE_REQUIRED_INCLUDES ${ICONV_INCLUDE_DIR} ${LIBXML2_INCLUDE_DIR})
-  CHECK_INCLUDE_FILES("libxml/xmlreader.h" HAVE_LIBXML_XMLREADER_H)
-  SET(CMAKE_REQUIRED_INCLUDES "")
-ELSE(LIBXML2_FOUND)
-  #
-  # Find Expat
-  #
-  FIND_PACKAGE(EXPAT)
-  IF(EXPAT_FOUND)
-    INCLUDE_DIRECTORIES(${EXPAT_INCLUDE_DIR})
-    LIST(APPEND ADDITIONAL_LIBS ${EXPAT_LIBRARIES})
-    SET(HAVE_LIBEXPAT 1)
-    LA_CHECK_INCLUDE_FILE("expat.h" HAVE_EXPAT_H)
-  ENDIF(EXPAT_FOUND)
-ENDIF(LIBXML2_FOUND)
+IF (ENABLE_XML)
+  FIND_PACKAGE(LibXml2)
+
+  IF(LIBXML2_FOUND)
+    INCLUDE_DIRECTORIES(${LIBXML2_INCLUDE_DIR})
+    LIST(APPEND ADDITIONAL_LIBS ${LIBXML2_LIBRARIES})
+    SET(HAVE_LIBXML2 1)
+    # libxml2's include files use iconv.h
+    # We need a directory path of iconv.h so that it won't fail to check
+    # "libxml/xmlreader.h".
+    FIND_PATH(ICONV_INCLUDE_DIR iconv.h)
+    SET(CMAKE_REQUIRED_INCLUDES ${ICONV_INCLUDE_DIR} ${LIBXML2_INCLUDE_DIR})
+    CHECK_INCLUDE_FILES("libxml/xmlreader.h" HAVE_LIBXML_XMLREADER_H)
+    SET(CMAKE_REQUIRED_INCLUDES "")
+  ELSE(LIBXML2_FOUND)
+    #
+    # Find Expat
+    #
+    FIND_PACKAGE(EXPAT)
+    IF(EXPAT_FOUND)
+      INCLUDE_DIRECTORIES(${EXPAT_INCLUDE_DIR})
+      LIST(APPEND ADDITIONAL_LIBS ${EXPAT_LIBRARIES})
+      SET(HAVE_LIBEXPAT 1)
+      LA_CHECK_INCLUDE_FILE("expat.h" HAVE_EXPAT_H)
+    ENDIF(EXPAT_FOUND)
+  ENDIF(LIBXML2_FOUND)
+ELSE (ENABLE_XML)
+  SET(EXPAT_FOUND 0)
+  SET(LIBXML2_FOUND 0)
+ENDIF (ENABLE_XML)
 
 #
 # Check functions
Only in libarchive-2.8.0: CMakeLists.txt~
