--- openssl-1.0.0/Configure.orig	2010-04-26 10:54:47.000000000 -0700
+++ openssl-1.0.0/Configure	2010-04-26 10:55:55.000000000 -0700
@@ -1409,6 +1409,8 @@
 	}
 $cmll_obj=$cmll_enc	unless ($cmll_obj =~ /.o$/);
 
+$cflags .= " " . $ENV{BP_EXTRA_CFLAGS};
+
 # "Stringify" the C flags string.  This permits it to be made part of a string
 # and works as well on command lines.
 $cflags =~ s/([\\\"])/\\\1/g;
